src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseUtilities.hpp

```diff
@@ -28,31 +28,36 @@ namespace compositionalMultiphaseUtilities
 {
 
 /**
- * @brief Shifts all elements one position ahead and replaces the first element
- * with the sum of all elements in each block of the input block one-dimensional
- * array of values.
+ * @brief In each block, shift the elements from 0 to numRowsToShift-1 one position ahead and replaces the first element
+ * with the sum of all elements from 0 to numRowsToShift-1 of the input block one-dimensional array of values.
  *
  * @tparam VEC type of one-dimensional array of values
- * @param N block size
- * @param NB number of blocks
+ * @param numRowsToShift number of rows to shift in each block
+ * @param numRowsInBlock block size
+ * @param numBlocks number of blocks
  * @param v block one-dimensional array of values
+ *
+ * @detail numRowsToShift is different from numRowsInBlock if there is an equation that we do *not* want to
+ * modify, like the energy flux for thermal simulations. In that specific case, numRowsToShift
+ * is set to numRowsInBlock-1 to make sure that we don't modify the last row of each block (the energy flux)
  */
 template< typename VEC >
 GEOSX_HOST_DEVICE
-void shiftBlockElementsAheadByOneAndReplaceFirstElementWithSum( integer const N,
-                                                                integer const NB,
+void shiftBlockElementsAheadByOneAndReplaceFirstElementWithSum( integer const numRowsToShift,
```

Comment:
I am not sure I fully understand what's going on here.

Reply:
Here I modified the code shifting the rows and computing the row sum of mass equations because when there is an energy equation, we don't want to include it in the sum, and we don't want to shift it.

In the latest commit, I tried to document what the function does with more doxygen that explains the thermal case:

The `localFluxJacobian` looks like that:
```
 *                       p    \rho_1   \rho_2   T
 *    (mass_comp_1)_1    .       .       .      .       \
 *    (mass_comp_2)_1    .       .       .      .       |=> block 1
 *    (energy)_1         .       .       .      .       /
 *
 *    (mass_comp_1)_2    .       .       .      .       \
 *    (mass_comp_2)_2    .       .       .      .       |=> block 2
 *    (energy)_2         .       .       .      .       /
 ```
and we have
```
 *      - numRowsToShift = 2
 *      - numRowsInBlock = 3
 *      - numColsInBlock = 4
 *      - numBlocks = 2
 ```
So, in each block, the code is going to sum the `mass_comp_X` equations, shift them, but will just skip the `energy` equation. This was the reason for adding the argument `numRowsToShift` to the function.